pythonforandroid/recipes/python2/__init__.py

```diff
@@ -56,89 +116,122 @@ def build_arch(self, arch):
         if not exists(join(self.ctx.get_libs_dir(arch.arch), 'libpython2.7.so')):
             shprint(sh.cp, join(self.get_build_dir(arch.arch), 'libpython2.7.so'), self.ctx.get_libs_dir(arch.arch))
 
-
-        # # if exists(join(self.get_build_dir(arch.arch), 'libpython2.7.so')):
-        # if exists(join(self.ctx.libs_dir, 'libpython2.7.so')):
-        #     info('libpython2.7.so already exists, skipping python build.')
-        #     if not exists(join(self.ctx.get_python_install_dir(), 'libpython2.7.so')):
-        #         info('Copying python-install to dist-dependent location')
-        #         shprint(sh.cp, '-a', 'python-install', self.ctx.get_python_install_dir())
-        #     self.ctx.hostpython = join(self.ctx.get_python_install_dir(), 'bin', 'python.host')
-
-        #     return
-
     def do_python_build(self, arch):
-        if 'sqlite' in self.ctx.recipe_build_order:
-            print('sqlite support not yet enabled in python recipe')
-            exit(1)
-
-        hostpython_recipe = Recipe.get_recipe('hostpython2', self.ctx)
         shprint(sh.cp, self.ctx.hostpython, self.get_build_dir(arch.arch))
-        shprint(sh.cp, self.ctx.hostpgen, self.get_build_dir(arch.arch))
+        shprint(sh.cp, self.ctx.hostpgen, join(self.get_build_dir(arch.arch), 'Parser'))
         hostpython = join(self.get_build_dir(arch.arch), 'hostpython')
-        hostpgen = join(self.get_build_dir(arch.arch), 'hostpython')
+        hostpgen = join(self.get_build_dir(arch.arch), 'Parser/hostpgen')
 
         with current_directory(self.get_build_dir(arch.arch)):
-
-
             hostpython_recipe = Recipe.get_recipe('hostpython2', self.ctx)
             shprint(sh.cp, join(hostpython_recipe.get_recipe_dir(), 'Setup'), 'Modules')
+            # Hack to make it work from user recipes, referenced on python-for-android issues  #613
+            # shprint(sh.cp, join(hostpython_recipe.recipe_dir, 'Setup'), 'Modules/Setup')
 
             env = arch.get_env()
-
-            # AND: Should probably move these to get_recipe_env for
-            # neatness, but the whole recipe needs tidying along these
-            # lines
-            env['HOSTARCH'] = 'arm-eabi'
-            env['BUILDARCH'] = shprint(sh.gcc, '-dumpmachine').stdout.decode('utf-8').split('\n')[0]
-            env['CFLAGS'] = ' '.join([env['CFLAGS'], '-DNO_MALLINFO'])
+            env.update({
+                'RFS': "{0}/platforms/android-{1}/arch-arm".format(self.ctx.ndk_dir, self.ctx.android_api),
+                })
+            # info('\t-> RFS:               {}'.format(env['RFS']))
```

Comment:
Ok, you are right...already removed from my local source code and I will push changes later